src/App.jsx

```diff
@@ -14,45 +15,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABETICALLY = 'alphabetically';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(goods, sortField, reverseField) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState(false);
+
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    sortField,
+    reverseField,
+  );
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_FIELD_ALPHABETICALLY)}
+          type="button"
+          className={`button ${sortField === SORT_FIELD_ALPHABETICALLY ? 'is-info' : 'is-info is-light'}`}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          type="button"
+          className={`button ${sortField === SORT_FIELD_LENGTH ? 'is-success' : 'is-success is-light'}`}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setReverseField(!reverseField)}
+          type="button"
+          className={`button ${reverseField ? 'is-warning' : 'is-warning is-light'}`}
+        >
+          Reverse
+        </button>
+
+        {(sortField || reverseField) && (
+          <button
+            onClick={() => {
+              setSortField('');
+              setReverseField(false);
+            }}
```

Comment:
It's better to create separate functions to handle events. This way, your code will be easier to maintain and change
```suggestion
            onClick={onReset}
```